### PR TITLE
Rc.1

### LINKED ISSUE #681: Terms & Conditions configuration
Add configuration options to support optional terms & conditions ui

### LINKED ISSUE #710: Keyboard Accessibility for Modal Dialog
**Is your feature request related to a problem? Please describe.**
Currently, when opening the Web3Modal dialog via keyboard, focus is left behind on the activating element. A user may tab through the rest of the page to eventually reach the inside of the modal dialog, but this is not ideal.

**Describe the solution you'd like**
Consistent with WAI-ARIA best practices, opening the Web3Modal dialog should move focus to the dialog. Focus should then be trapped within the modal dialog until the modal is dismissed, via clicking outside, clicking close, or via the <kbd>Esc</kbd> key. When dismissing, an attempt should be made to restore focus to the element previously focused before the modal dialog was opened.

**Describe alternatives you've considered**
N/A

**Additional context**
WAI-ARIA Modal Dialog Pattern: https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/

### LINKED ISSUE #714: Web3Modal V2 Beta.5 iOS, testing mobile browser issues.
**Context**
Using `export const ethereumClient = new EthereumClient(wagmiClient, [])` with empty chains array because the dApp is a bridge which needs to switch to multiple chains from the UI depending on the bridge selected rather than selecting a network from the modal.
Using `useSigner` to call contracts:
```
const { data } = useSigner()
const web3Provider: ethers.providers.Web3Provider = data.provider
```
All issues mentioned below work correctly from desktop browsers (connected to WalletConnect or Injected providers) and mobile wallet embedded browsers.

**Connecting from Safari Mobile (iOS)**
- The other installed wallets like MetaMask Mobile and Rainbow are not available by default but connecting to them from `View All` works.

- After signing a transaction from MetaMask Mobile and redirecting back to Safari, the contract call promise never resolved and the dApp stays in pending state waiting for the transaction hash.
Rejecting the prompt in MetaMask Mobile does work as expected and the contract call is rejected.
Network and account switching worked correctly.

**Connecting from Brave Mobile (iOS)**
- The Brave Mobile injected wallet is not available and cannot be connected to (same screenshot).
- The available wallets are the same as in Safari and connecting to MetaMask requires going through `View All`.
Similar to Safari, signing a transaction in MetaMask didn't resolve the promise.
Rejecting the MetaMask prompt also didn't resolve the promise unlike Safari which handled it correctly.

### LINKED ISSUE #718: Android mobile view (Connect button instead of wallet list)
## Context

Android and iOS work differently.

On iOS the user is shown a list of mobile browsers.
On Android, the user is shown a connect button, and then the OS will show a list of WalletConnect compatible wallets.

## TODO

This feature is in web3modal v1, but missing from v2.
Add it to v2.

### LINKED ISSUE #720: Chains select widget

### LINKED ISSUE #725: Explorer allow/deny list configuration
Customize what wallets are shown

`explorerAllowList` - Coming in beta.7, will allow to manage explorer records if projectId was provided
`explorerDenyList` - Coming in beta.7, will allow to manage explorer records if projectId was provided

### LINKED ISSUE #784: Customise modal's z-index
I need to set the level of the popover. How do I handle that

### LINKED ISSUE #785: Add "defaultChain" option for dapp developers
Allow developers to define initial (default) chain for user to connect with and skip network selection view if such is defined.

### LINKED ISSUE #793: Deeplink parsing breaks deeplinks with nested paths
**Describe the bug**
"When I try to connect to OKEx using the web3modal, I am unable to redirect to the OKEx app
**SDK Version**
 - Version [2.0.0-alpha.10]

**Screenshots**
4541671762721_ pic

image

### PR DESCRIPTION
**Pull request by 0xAsimetriq**
## Changes
- closes #718 (android mobile view)
- closes  #784 (added `themeZIndex` config)
- closes #785 (added `defaultChain` config)
- closes #720 (added network switch component)
- closes #725 (add `explorerAllowList` and `explorerDenyList`)  
- closes #710 (Modal focus management)
- closes #793 (Account for deeplinks with nested paths when parsing / cleaning them up)
- closes #714 (No wallets displayed when empty `[]` chains provided to EthereumClient)
- closes #681 (Add configs for terms and policy)

## Breaking Changes
- Changed `enableNetworkView` config option to be `false` by default (was `true`)

## TODO
- Update web3modal.com

### CODE REVIEW

File: packages/core/src/controllers/ConfigCtrl.ts [hunk 1]
```diff
@@ -13,14 +14,20 @@ const state = proxy<ConfigCtrlState>({
   themeMode: isDarkMode() ? 'dark' : 'light',
   themeColor: 'default',
   themeBackground: CoreUtil.isMobile() ? 'themeColor' : 'gradient',
+  themeZIndex: 89,
```
Comment (crypblizz8) [inline packages/core/src/controllers/ConfigCtrl.ts:17]: Non Blocking: Can you explain how they overrides the whole zIndex PR you are closing?

Comment (0xAsimetriq) [reply packages/core/src/controllers/ConfigCtrl.ts:17]: We default to z-index of `89` but there is now an option to override it. Override happens in `ThemeUtil` file within `setTheme` method (using css var)

**Review by crypblizz8 — APPROVED** [review]
LGTM, happy to see all these final features come through!

**Review by devceline — APPROVED** [review]
LGTM

**Review by devceline — APPROVED** [review]
LGTM

**Pull request merged**